test/solidity_dynamic_strings.js

```diff
@@ -0,0 +1,54 @@
+const Web3 = require("web3");
+const assert = require("assert");
+const Ganache = require("../index");
+const path = require("path");
+const compileAndDeploy = require("./helpers/contracts").compileAndDeploy;
+
+const mnemonic = "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat";
+
+function setUp(options = { mnemonic }, contractName = "DynamicStringLength") {
+  const context = {
+    options: options,
+    provider: null,
+    web3: null,
+    accounts: [],
+    contractArtifact: {},
+    instance: null
+  };
+
+  before("setup web3", async function() {
+    context.provider = Ganache.provider(context.options);
+    context.options.blockTime = 2000;
+    context.web3 = new Web3(context.provider);
+  });
+
+  before("compile source", async function() {
+    this.timeout(10000);
+    context.contractArtifact = await compileAndDeploy(
+      path.join(__dirname, ".", `${contractName}.sol`),
+      contractName,
+      context.web3
+    );
+    context.instance = context.contractArtifact.instance;
```

Comment:
unless you're using `contractArtifact`, I'd drop `contractArtifact` altogether and do

```
const { instance } = await compileAndDeploy( ...
```

^ sweet ES6 destructuringing syntax will pull the `instance` field out of the object returned by `compileAndDeploy` and assign it to a const variable of the same name.

Reply:
Great suggestions for overall refactoring efforts.